cmd/gardenlet/app/migration.go

```diff
@@ -36,6 +44,43 @@ func (g *garden) runMigrations(ctx context.Context, log logr.Logger, _ cluster.C
 	return nil
 }
 
+// TODO(timuthy): Remove after v1.96 was released.
+// updateGardenerResourceManager updates all GRM deployments in a seed cluster,
+// in order to roll out the Brotli compression support added in Gardener v1.96.
+func updateGardenerResourceManager(ctx context.Context, log logr.Logger, cl client.Client) error {
```

Comment:
Do I get this right that we want to update all shoot/seed resource managers? What about scenarios where garden and seed run in the same cluster? What should happen there? It seems like we would also update the "ordinary" resource manager in garden namespace, but may not update specialized ones, e.g. resource manager for the virtual garden cluster in the operator scenario.

Reply:
Very nice question!
The GRM deployments under the control of `gardener-operator` shouldn't be patched because it already updates it before the new `gardenlet`s are rolled out.
It's safer to skip handling of the GRM in the Runtime Garden cluster here, otherwise we might even risk a downgrade or conflicts, when `gardener-operator` is already updated but `gardenlet` restarts on an older version.